lib/providers/config_provider.dart

```diff
@@ -1,142 +1,370 @@
-        .toList();
+    return getSortedWidgetsList(currentConnection.favouriteWidgets);
   }
 
   List<DashboardWidget> get quarantineWidgets {
-    return _config.widgets.widgetsById.entries
-        .map((entry) => entry.value)
-        .where((widget) => _urlPreferences.quarantineWidgetIds.contains(widget.id))
-        .toList();
+    return getSortedWidgetsList(currentConnection.quarantineWidgets);
+  }
+
+  bool isWidgetInQuarantine(DashboardWidget widget) {
+    return quarantineWidgets.indexWhere((element) => element.id == widget.id) > -1;
   }
 
   List<Board> get boards {
     return [..._boards];
   }
 
-  Future<void> fetchConfig() async {
-    final response = await http.get(_currentUrl);
-    _config = Config.fromJson(json.decode(response.body) as Map<String, dynamic>);
-    _boards = _config.boards.boardsById.entries.map((entry) => entry.value).toList();
-    if (await SharedPref.containsKey(_currentUrl)) {
-      _urlPreferences = UrlPreferences.fromJson(jsonDecode(await SharedPref.read(_currentUrl)));
-    }
-    notifyListeners();
+  List<DashboardWidget> getAllWidgets() {
+    return _config.widgets.widgetsById.entries.map((entry) => entry.value).toList();
+  }
+
+  List<DashboardWidget> getAllWidgetsDeepCopy() {
+    return new List<DashboardWidget>.from(getAllWidgets().map((widget) => DashboardWidget.deepCopy(widget)).toList());
+  }
+
+  String getWidgetName(DashboardWidget widget) {
+    return (widget.title != null && widget.title.isNotEmpty) ? widget.title : widget.type;
   }
 
   List<DashboardWidget> getBoardWidgets(Board board) {
-    return _config.widgets.widgetsById.entries
-        .map((entry) => entry.value)
-        .where((widget) => board.widgets.contains(widget.id))
-        .toList();
+    List<DashboardWidget> boardWidgets = [
+      ..._config.widgets.widgetsById.entries
+          .map((entry) => entry.value)
+          .where((widget) =>
+              board.widgets.contains(widget.id) &&
+              quarantineWidgets.indexWhere((element) => element.id == widget.id) == -1 &&
+              widget.type != "WhiteSpaceWidget")
+          .toList()
+    ];
+    return getSortedWidgetsList(boardWidgets);
   }
 
-  void updateWidget(Map<String, dynamic> widgetData) {
-    this._config.widgets.widgetsById.forEach((key, value) {
-      if (key == widgetData['id']) {
-        value.updateWidget(widgetData);
-        notifyListeners();
+  List<DashboardWidget> getSortedWidgetsList(List<DashboardWidget> boardWidgets) {
+    switch (_settingsPreferences.sortBy) {
+      case WidgetSortTypes.NONE:
+        return boardWidgets;
+      case WidgetSortTypes.NAME_ASCENDING:
+        boardWidgets.sort((a, b) => getWidgetName(a).compareTo(getWidgetName(b)));
+        break;
+      case WidgetSortTypes.NAME_DESCENDING:
+        boardWidgets.sort((a, b) => getWidgetName(a).compareTo(getWidgetName(b)));
+        boardWidgets = boardWidgets.reversed.toList();
```

Comment:
probably this should work:
`boardWidgets.sort((a, b) => getWidgetName(b).compareTo(getWidgetName(a)));`
and this is not required then: `boardWidgets.reversed.toList();`

Reply:
done

Reply:
done